### PR TITLE
feat: add Dropout layer support to backend

### PR DESCRIPTION
**Pull request by paramananda-15**
## Description

Adds support for the Dropout layer in the TensorMap backend model generation service.

This change allows ReactFlow graphs to include Dropout nodes that are translated into
`tf.keras.layers.Dropout` when constructing the Keras functional model.

The implementation adds handling for a new node type (`customdropout`) in
`app/services/model_generation.py`, enabling TensorMap to support dropout-based
regularization in generated deep learning models.

This contribution aligns with the TensorMap GSoC project goal of expanding the
supported layer types in the backend.

Fixes #N/A

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

## How Has This Been Tested?

All existing backend tests were executed locally after implementing the change
to ensure no regressions were introduced.

Command used:
python -m pytest

Results:
130 passed in ~3 seconds

- [x] Existing tests pass
- [ ] New tests added
- [x] Manual testing

## Screenshots (if applicable)

N/A

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review
- [ ] I have added/updated documentation as needed
- [x] My changes generate no new warnings
- [x] Tests pass locally

### CODE REVIEW

**Comment by paramananda-15** [comment]
Hi! I'm interested in contributing to TensorMap for GSoC and started with this small feature addition to support Dropout layers in the backend model generation pipeline. Looking forward to feedback and suggestions for further improvements.

**Comment by Sickwoman** [comment]
Hi @paramananda-15 , This layer has already got merged not just the backend but the full stack. 
Work on different layers either. 
Happy coding

**Comment by paramananda-15** [comment]
> Hi @paramananda-15 , This layer has already got merged not just the backend but the full stack. Work on different layers either. Happy coding

Thanks for the clarification! I’ll close this PR since the Dropout layer has already been implemented. I’ll look into contributing other layers or improvements mentioned in the project roadmap. Looking forward to contributing more.
